crates/sui-bridge/src/sui_transaction_builder.rs

```diff
@@ -0,0 +1,225 @@
+            "Failed to serialize source_chain: {:?}. Err: {:?}",
+            source_chain, e
+        ))
+    })?;
+    let seq_num = builder.pure(seq_num).map_err(|e| {
+        BridgeError::BridgeEventParameterSerializationError(format!(
+            "Failed to serialize seq_num: {:?}. Err: {:?}",
+            seq_num, e
+        ))
+    })?;
+    let sender = builder.pure(sender.clone()).map_err(|e| {
+        BridgeError::BridgeEventParameterSerializationError(format!(
+            "Failed to serialize sender: {:?}. Err: {:?}",
+            sender, e
+        ))
+    })?;
+    let target_chain = builder.pure(target_chain as u8).map_err(|e| {
+        BridgeError::BridgeEventParameterSerializationError(format!(
+            "Failed to serialize target_chain: {:?}. Err: {:?}",
+            target_chain, e
+        ))
+    })?;
+    let target = builder.pure(target.clone()).map_err(|e| {
+        BridgeError::BridgeEventParameterSerializationError(format!(
+            "Failed to serialize target: {:?}. Err: {:?}",
+            target, e
+        ))
+    })?;
+    let arg_token_type = builder.pure(token_type as u8).map_err(|e| {
+        BridgeError::BridgeEventParameterSerializationError(format!(
+            "Failed to serialize token_type: {:?}. Err: {:?}",
+            token_type, e
+        ))
+    })?;
+    let amount = builder.pure(amount).map_err(|e| {
+        BridgeError::BridgeEventParameterSerializationError(format!(
+            "Failed to serialize amount: {:?}. Err: {:?}",
+            amount, e
+        ))
+    })?;
+
+    let arg_msg = builder.programmable_move_call(
+        *get_bridge_package_id(),
+        ident_str!("message").to_owned(),
+        ident_str!("create_token_bridge_message").to_owned(),
+        vec![],
+        vec![
+            source_chain,
+            seq_num,
+            sender,
+            target_chain,
+            target,
+            arg_token_type,
+            amount,
+        ],
+    );
+
+    // Unwrap: this should not fail
+    let arg_bridge = builder.obj(*get_root_bridge_object_arg()).unwrap();
+
+    let mut sig_bytes = vec![];
+    for (_, sig) in sigs.signatures {
+        sig_bytes.push(sig.as_bytes().to_vec());
+    }
+    let arg_signatures = builder.pure(sig_bytes).unwrap();
+
+    builder.programmable_move_call(
+        *get_bridge_package_id(),
+        ident_str!("bridge").to_owned(),
+        ident_str!("approve_bridge_message").to_owned(),
+        vec![],
+        vec![arg_bridge, arg_msg, arg_signatures],
+    );
+
+    builder.programmable_move_call(
+        *get_bridge_package_id(),
+        ident_str!("bridge").to_owned(),
+        ident_str!("claim_and_transfer_token").to_owned(),
+        vec![get_sui_token_type_tag(token_type)],
+        vec![arg_bridge, source_chain, seq_num],
+    );
+
+    let pt = builder.finish();
+
+    Ok(TransactionData::new_programmable(
+        client_address,
+        vec![*gas_object_ref],
+        pt,
+        15_000_000,
+        1500,
```

Comment:
Hard coding gas price is going to be fragile.

Reply:
good catch

Reply:
this will require adding a new function in some traits. Added TODOs, will fix in follow up PRs